aql-engine/src/test/java/org/ehrbase/openehr/aqlengine/service/AqlQueryServiceImpTest.java

```diff
@@ -55,4 +101,51 @@ void resolveEhrCompositions(String srcAql, String expectedAql) {
         AqlQueryServiceImp.replaceEhrPaths(aqlQuery);
         assertThat(aqlQuery.render()).isEqualTo(expectedAql.replaceAll(" +", " "));
     }
+
+    @ParameterizedTest
+    @CsvSource(
+            textBlock =
+                    """
+                5||10||Invalid AQL query: fetch is defined on query 5 and as parameter 10
+                5|20||40|Invalid AQL query: offset is defined on query 20 and as parameter 40
+                5|||30|Invalid AQL query: limit is defined on query 5 and offset as parameter 30
+                |||42|Invalid AQL query: provided offset 42 without a limit
```

Comment:
@HolgerReiseVSys there are currently 4 different message used, base on the given combination. Not sure, but possible we could combine them into one generic message.

Reply:
Maybe it would be sufficient to state that because the query contains a LIMIT clause, fetch and offset parameters must not be used

Reply:
Reduced to 2 messages -> https://github.com/ehrbase/ehrbase/pull/1325/files#diff-92431608f1168dac98145160a7f45a69eeecd53d858d5683646664c86f7a83dfR109-R112